reports/diversification.py

```diff
@@ -1,27 +1,33 @@
+import datetime
+from collections import defaultdict
+import pandas
 from logger import logger
+from marketplace import marketplace
 from . import calculator
 
 
-def generate_report_per_date(overall_report):
-    diversification_report_per_date = {}
-    for date, overall_data in overall_report.items():
-        diversification_report_per_date[date] = generate_report(date, overall_data)
+def generate_report_per_date(df_investiments):
+    diversification_report_per_date = defaultdict(datetime.datetime)
+    for date in sorted(df_investiments[marketplace.FILE_DATE].unique()):
```

Comment:
Why sorted in this case? It is a dictionare and it is irrelevant the order that you use to insert the items. If you would want to serialize it to json then it would make sense because you want to have the same order every time.

Reply:
You are right, but we must change the print report function accordingly. Moreover, I don't know if we also will have to change the tests.

If you don't think that this is critical, I would prefer not to change it now. @limdor Could it be better to create a new issue to change both tests simultaneously?